sql/src/main/java/io/crate/analyze/relations/PlannedAnalyzedRelation.java

```diff
@@ -30,12 +31,13 @@
     /**
      * Returns a plan for this relation.
      */
-    public Plan plan();
+    Plan plan();
 
-    public void addProjection(Projection projection);
+    void addProjection(Projection projection);
 
-    public boolean resultIsDistributed();
+    boolean resultIsDistributed();
 
-    public DQLPlanNode resultNode();
+    DQLPlanNode resultNode();
 
+    UpstreamPhase lastUpstreamPhase();
```

Comment:
try to solve this by setting a var on ConsumerContext see ConsumerContext.requiredPageSize as an example